### PR TITLE
Completed changes for task 4

### PR DESCRIPTION
**Pull request by monika-proczek**

### CODE REVIEW

File: force-app/main/default/triggers/VehicleToSalonTrigger.trigger [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    Set<Id> listOfVehiclesIds = new Set<Id>();
+    List<Vehicle_to_Salon__c> offersAvailableInSalon = [SELECT Id, Vehicle__c FROM Vehicle_to_Salon__c WHERE Car_Availibility__c='Available in Salon'];
+
+    for (Vehicle_to_Salon__c existingOffer: offersAvailableInSalon){
+        listOfVehiclesIds.add(existingOffer.Vehicle__c);
+    }
+
+    for(Vehicle_to_Salon__c newOffer : Trigger.new){
+        if(newOffer.Car_Availibility__c=='Available in Salon' && listOfVehiclesIds.contains(newOffer.Vehicle__c)){
+            newOffer.addError('You cannot create such an offer because such a vehicle is already available in the salon through an existing offer.');
```
Comment (AdrianDlu) [inline force-app/main/default/triggers/VehicleToSalonTrigger.trigger:12]: please show on what salon this car is avilable

**Review by AdrianDlu — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/TestVehicleToSalonTrigger.cls [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    static void makeData(){
+        Vehicle__c newBMW = TestDataFactory.createVehicles(1,'new BMW X5')[0];
+        Account newAccount = TestDataFactory.createAccounts(1, 'test account')[0];
+        Salon__c newSalon = TestDataFactory.createSalons(1, 'newTestSalon', newAccount.Id, 'PassengerCars', 'Closed')[0];
+        TestDataFactory.createSingleOffer('new BMW X5', newBMW.Id, newSalon.Id, 'Available in Salon');
+    }
+    
+    @isTest
+    static void testIfOfferIsPrevented() {
+        Vehicle__c existingVehicle = [SELECT Id FROM Vehicle__c WHERE Vehicle_Identification_Number__c = '17849284653746280'][0];
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/classes/TestVehicleToSalonTrigger.cls:14]: Done

File: force-app/main/default/classes/TestVehicleToSalonTrigger.cls [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        Vehicle_to_Salon__c newOffer = new Vehicle_to_Salon__c(Name = 'new BMW X5 in other salon',
+        Vehicle__c = existingVehicle.Id,
+        Salon__c = salon.Id,
+        Car_Availibility__c = 'Available in Salon');
+
+        Test.startTest();
+        Database.SaveResult result = Database.insert(newOffer, false);
+        Test.stopTest();
+
+        List<Vehicle_to_Salon__c> newOffersList = [SELECT Id FROM Vehicle_to_Salon__c WHERE Name = 'new BMW X5 in other salon'];
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/classes/TestVehicleToSalonTrigger.cls:25]: Done

File: force-app/main/default/triggers/VehicleToSalonTrigger.trigger (hunk 1 above)
Comment (monika-proczek) [reply force-app/main/default/triggers/VehicleToSalonTrigger.trigger:12]: done

File: force-app/main/default/classes/TestDataFactory.cls [hunk 4]
```diff
@@ -0,0 +1,55 @@
+@isTest
+public class TestDataFactory {
+    
+    public static List<Vehicle__c> createVehicles(Integer numVehicles, String commonName) {
+        List<Vehicle__c> listOfVehicles = new List<Vehicle__c>();
+        
+        for(Integer i=0;i<numVehicles;i++) {
+            Vehicle__c newVehicle = new Vehicle__c(Name = commonName,
+            Vehicle_Identification_Number__c = '1784928465374628' + i,
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/classes/TestDataFactory.cls:9]: updated to single car

File: force-app/main/default/classes/TestDataFactory.cls [hunk 5]
```diff
@@ -0,0 +1,55 @@
+@isTest
+public class TestDataFactory {
+
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/classes/TestDataFactory.cls:3]: done

File: force-app/main/default/classes/TestDataFactory.cls [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        List<Vehicle__c> listOfVehicles = new List<Vehicle__c>();
+        
+        for(Integer i=0;i<numVehicles;i++) {
+            Vehicle__c newVehicle = new Vehicle__c(Name = commonName,
+            Vehicle_Identification_Number__c = '1784928465374628' + i,
+            Automotive_Brand__c = 'BMW');
+            listOfVehicles.add(newVehicle);
+        }
+        
+        insert listOfVehicles;
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/classes/TestDataFactory.cls:14]: sure, will change it in the future

File: force-app/main/default/classes/TestVehicleToSalonTrigger.cls [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        Account newAccount = TestDataFactory.createAccounts(1, 'test account')[0];
+        Salon__c newSalon = TestDataFactory.createSalons(1, 'newTestSalon', newAccount.Id, 'PassengerCars', 'Closed')[0];
+        TestDataFactory.createSingleOffer('new BMW X5', newBMW.Id, newSalon.Id, 'Available in Salon');
+    }
+    
+    @isTest
+    static void testIfOfferIsPrevented() {
+        Vehicle__c existingVehicle = [SELECT Id FROM Vehicle__c WHERE Vehicle_Identification_Number__c = '17849284653746280'][0];
+        Salon__c salon = [SELECT Id FROM Salon__c WHERE Name = 'newTestSalon0'][0];
+        Vehicle_to_Salon__c newOffer = new Vehicle_to_Salon__c(Name = 'new BMW X5 in other salon',
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/classes/TestVehicleToSalonTrigger.cls:16]: it is updated to different salon

File: force-app/main/default/triggers/VehicleToSalonTrigger.trigger [hunk 8]
```diff
@@ -0,0 +1,15 @@
+trigger VehicleToSalonTrigger on Vehicle_to_Salon__c (before insert) {
+   
+    Set<Id> listOfVehiclesIds = new Set<Id>();
+    List<Vehicle_to_Salon__c> offersAvailableInSalon = [SELECT Id, Vehicle__c FROM Vehicle_to_Salon__c WHERE Car_Availibility__c='Available in Salon'];
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/triggers/VehicleToSalonTrigger.trigger:4]: done

File: force-app/main/default/classes/TestVehicleToSalonTrigger.cls [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        Vehicle__c = existingVehicle.Id,
+        Salon__c = salon.Id,
+        Car_Availibility__c = 'Available in Salon');
+
+        Test.startTest();
+        Database.SaveResult result = Database.insert(newOffer, false);
+        Test.stopTest();
+
+        List<Vehicle_to_Salon__c> newOffersList = [SELECT Id FROM Vehicle_to_Salon__c WHERE Name = 'new BMW X5 in other salon'];
+        System.assertEquals(0, newOffersList.size());
```
[the remark this answers is not in the document]
Comment (monika-proczek) [reply force-app/main/default/classes/TestVehicleToSalonTrigger.cls:26]: sure, will remember

**Review by AdrianDlu — APPROVED** [review]

**Pull request merged**